src/InputField.js

```diff
@@ -36,29 +36,27 @@ const getInputStyles = showLabel => {
       }
 }
 
-class InputField extends React.Component {
-  constructor(props) {
-    super(props)
-
-    let hasInitialValue
-
-    React.Children.forEach(props.children, child => {
-      if (child && child.type === Input) {
-        hasInitialValue = !!child.props.value
-      }
-    })
+const noop = () => {}
 
-    this.state = {
-      showLabel: hasInitialValue
-    }
+class InputField extends React.Component {
+  static defaultProps = {
+    // for backwards-compatibility
+    onChange: noop
   }
 
-  onInputChange = event => {
-    this.setState({
-      showLabel: event.target.value
-    })
+  // for backwards-compatibility
+  handleChange = onChange => e => {
+    this.props.onChange(e)
+    if (typeof onChange !== 'function') return
+    onChange(e)
+  }
 
-    this.props.onChange(event)
+  hasValue = () => {
+    const { children } = this.props
+    return React.Children.toArray(children).reduce(
```

Comment:
Trivial, but if only `children` prop is destructed from `this.props` & used, (instead of multiple destructed props), is there any benefit about the destructing pattern here over the dot-notation object access, such as using `this.props.children` directly in the array `.reduce` call? (Is it for expressive, typing-less reason for `children`?)
```
return React.Children.toArray(this.props.children).reduce( ... )
```